architecture/additional_concepts/dynamic_admission_controllers.adoc

```diff
@@ -0,0 +1,473 @@
++
+After you create the `initializerConfiguration`, the system will take a few
+seconds to honor the new configuration. 
+
+. {product-title} appends the name of the initializer configuration object, here  `podimage.example.com`, 
+to the `metadata.initializers.pending` field of any newly created pods. 
+
+. An initializer controller should list and watch for uninitialized objects, by
+using the query parameter `includeUninitialized=true`. If using client-go, just
+set 
+[listOptions.includeUninitialized](https://github.com/kubernetes/kubernetes/blob/v1.7.0-rc.1/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/types.go#L315)
+to true.
+
+////
+
+[[architecture-additional-concepts-dynamic-admission-webhooks]]
+== External Admission Webhooks
+
+External admission webhooks are HTTP callbacks that receive
+admission requests and do something with them. What an external admission
+webhook does is up to you, but there is an
+link:https://github.com/kubernetes/kubernetes/blob/v1.7.0-rc.1/pkg/apis/admission/v1alpha1/types.go[interface]
+that it must adhere to so that it responds with whether or not the
+admission request should be allowed. 
+
+A simple example use case for an external admission webhook is semantic validation
+of resources. For example, you have an infrastructure that requires that all pods
+have a common set of labels and you do not want any pod to be
+persisted if the pod does not have those label. You could write your
+external admission webhook to do this validation before the pod is instantiated 
+and prevent the pod from being created.
+
+Some common use-cases include:
+
+* Mutating resources, for example such as to inject side-car containers into pods. 
+* Restricting projects, for example, to block some resources from a project. 
+* Complex CustomResource validation, such as performing complex validation on dependent fields.
+* Security response, for example, preventing certain SHAs from running.
+
+External admission webhooks allow a cluster administrator to add mutating and validating admission plugins 
+to the admission chain of the API server recompiling.
+
+[IMPORTANT]
+====
+*External Admission Webhooks* are Technology Preview featuresonly.
+ifdef::openshift-enterprise[]
+Technology Preview features are not supported with Red Hat production service
+level agreements (SLAs), might not be functionally complete, and Red Hat does
+not recommend to use them for production. These features provide early access to
+upcoming product features, enabling customers to test functionality and provide
+feedback during the development process.
+
+For more information on Red Hat Technology Preview features support scope, see
+https://access.redhat.com/support/offerings/techpreview/.
+endif::[]
+====
+
+The external admission webhooks support TLS because the admission control operations require high-security.
+
+Whenever a request comes in, the Generic Admission Webhook admission controller 
```

Comment:
this is about the validating webhooks. There are also mutating webhooks.

Reply:
The term of Generic Admission Webhook addmission controller is confusing IMO.

I shared a presentation with you. Those box graphics are maybe helpful to make this paragraph clearer.